src/mokernels/slfm.jl

```diff
@@ -0,0 +1,67 @@
+@doc raw"""
+    LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+
+A semiparametric kernel for problems involving multiple response variables.
+
+``k((x, p), (y, p)) = k_p(x, y) = \Sum^{Q}_{q=1} A_{pq}g_q(x, y) + e_p(x, y)``
+
+# Arguments
+- `g::AbstractVector{<:Kernel}`: an array of kernels
+- `e::AbstractVector{<:Kernel}`: an array of kernels
+- `A::AbstractMatrix`: an matrix of weights for the kernels of shape (length(e), length(g))
+
+
+# Reference:
+- [Seeger, Teh, and Jordan (2005)](https://infoscience.epfl.ch/record/161465/files/slfm-long.pdf)
+
+"""
+struct LatentFactorMOKernel{
+    Tg <:AbstractVector{<:Kernel}, 
+    Te <:AbstractVector{<:Kernel}, 
+    TA <: AbstractMatrix
+    } <: Kernel
+    g::Tg
+    e::Te
+    A::TA
+    function LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+        @assert (length(e), length(g)) == 
+            size(A) "Size of A not compatible to the given array of kernels"
+        return new{typeof(g), typeof(e), typeof(A)}(g, e, A)
+    end
+end
+
+function (κ::LatentFactorMOKernel)(x::Tuple{Vector, Int}, y::Tuple{Vector, Int})
+    if last(x) == last(y)
+        return sum([κ.g[i](first(x), first(y)) * κ.A[last(x), i] for i in 1:length(κ.g)]) + 
```

Comment:
Maybe avoid creating the intermediate array by writing
```suggestion
        return sum(κ.g[i](first(x), first(y)) * κ.A[last(x), i] for i in 1:length(κ.g)) + 
```
or (if the function arguments are changed)
```suggestion
        return sum(κ.g[i](x, y) * κ.A[px, i] for i in 1:length(κ.g)) + 
```